index.js

```diff
@@ -16,37 +16,67 @@ app.listen(PORT, function(req, res){
 	console.log(`Server is running @ ${PORT}`);
 })
 
+//End Date Func
+var date = new Date();
+var end = format(date);
+
+// middleware to get date
+var startDate = function(req,res,next){
+	axios.get(reqUrl,{ params:{
+		query: req.query.query,
+		sort: 'P_PDATE_D asc',
+		pageSize:1,
+		synonym: true,
+		format: 'json'  }})
+	.then(function(response){
+		 req.start = response.data.resultList.result[0].firstPublicationDate;
+		 next();		
+	})
+	.catch(function(err){
+	 	res.status(500).json({
+		 	content:'Api request failed'
+		})	
+	})
+}
+app.use(startDate);
 app.get('/getWithYear', malaria);
 
 function malaria(req, res){	
-		axios.get(reqUrl,{ params:{query: req.query.query,
-			date: `( FIRST_PDATE:[${req.query.start} TO ${req.query.end}])`,
-			sort: 'CITED desc',
-			pageSize:100,
-			synonym: true,
-			format: 'json' }})
-		.then(function(response){
-			var rest = {};
+	axios.get(reqUrl,{ params:{query: req.query.query,
```

Comment:
@Firnaa5 I suggest something like a variable use like `const params ={ the object here }` and use like
`axios.get(reqUrl, {params: params})` which again can be shorthanded with `axios.get(reqUrl, {params})` :slightly_smiling_face: